shared/js/l10n.js

```diff
@@ -1161,19 +1178,27 @@
     }
 
     var idToFetch = this.isPseudo ? ctx.defaultLocale : this.id;
+    var source = navigator.mozL10n._config.localeSources[this.id] || 'app';
+    var gaiaVersion = navigator.mozL10n._config.gaiaVersion;
+
     for (var i = 0; i < ctx.resLinks.length; i++) {
       var resLink = decodeURI(ctx.resLinks[i]);
       var path = resLink.replace('{locale}', idToFetch);
       var type = path.substr(path.lastIndexOf('.') + 1);
 
+      var cb = null;
+      var ioType = 'text';
       switch (type) {
         case 'json':
-          io.loadJSON(path, onJSONLoaded, sync);
+          ioType = 'json';
```

Comment:
I also think it's okay to simply call `bindingsIO[source]` under each `case`, with different arguments.